node/sc/bridge_manager.go

```diff
@@ -587,44 +587,44 @@ func (bm *BridgeManager) RestoreBridges() error {
 	bm.stopAllRecoveries()
 
 	for _, journal := range bm.journal.cache {
-		var cBridgeInfo *BridgeInfo
-		var pBridgeInfo *BridgeInfo
-		var ok bool
-
 		cBridgeAddr := journal.LocalAddress
 		pBridgeAddr := journal.RemoteAddress
 		bam := bm.subBridge.bridgeAccountManager
 
 		// Set bridge info
-		cBridgeInfo, ok = bm.GetBridgeInfo(cBridgeAddr)
-		if !ok {
-			cBridge, err := bridgecontract.NewBridge(cBridgeAddr, bm.subBridge.localBackend)
-			if err != nil || cBridge == nil {
-				logger.Error("local bridge creation is failed", "err", err, "bridge", cBridge)
-				break
-			}
-			err = bm.SetBridgeInfo(cBridgeAddr, cBridge, pBridgeAddr, nil, bam.scAccount, true, false)
+		cBridgeInfo, cOk := bm.GetBridgeInfo(cBridgeAddr)
+		pBridgeInfo, pOk := bm.GetBridgeInfo(pBridgeAddr)
+
+		cBridge, err := bridgecontract.NewBridge(cBridgeAddr, bm.subBridge.localBackend)
+		if err != nil {
+			logger.Error("local bridge creation is failed", "err", err, "bridge", cBridge)
+			break
+		}
+
+		pBridge, err := bridgecontract.NewBridge(pBridgeAddr, bm.subBridge.remoteBackend)
+		if err != nil {
+			logger.Error("remote bridge creation is failed", "err", err, "bridge", pBridge)
+			break
+		}
+
+		if !cOk {
+			err = bm.SetBridgeInfo(cBridgeAddr, cBridge, pBridgeAddr, pBridge, bam.scAccount, true, false)
 			if err != nil {
 				logger.Error("setting local bridge info is failed", "err", err)
 				bm.DeleteBridgeInfo(cBridgeAddr)
 				break
 			}
-			cBridgeInfo, ok = bm.GetBridgeInfo(cBridgeAddr)
+			cBridgeInfo, cOk = bm.GetBridgeInfo(cBridgeAddr)
```

Comment:
Why do we need to assign to cOk here? Is it used later?

Reply:
Yes.